thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java

```diff
@@ -155,12 +151,37 @@ public void bindNewView(@NonNull final V view) {
                             + "Binding a view is not allowed");
         }
 
-        if (view == null) {
-            throw new IllegalStateException(
-                    "the view cannot be set to null. Call #sleep() instead");
+        if (isViewAttached()) {
+            if (view.equals(mView)) {
+                TiLog.v(TAG, "not calling onAttachView(), view already attached");
+                return;
+            } else {
+                throw new IllegalStateException(
+                        "a view is already attached, call #detachView first");
+            }
+        }
+
+        if (!isInitialized()) {
+            throw new IllegalStateException("Presenter is not created, call #create() first");
         }
 
         mView = view;
+        moveToState(State.VIEW_ATTACHED, false);
+        mCalled = false;
```

Comment:
it's a general property for all super call checks. Implementation and naming copied from AOSP `Actiivty` and `Fragment` https://github.com/android/platform_frameworks_base/blob/4b1a8f46d6ec55796bf77fd8921a5a242a219278/core/java/android/app/Activity.java#L748